fog/sample-paykit/src/client.rs

```diff
@@ -362,6 +360,232 @@ impl Client {
         )
     }
 
+    /// Builds a signed contingent input that offers to trade "this" amount
+    /// for "that" amount.
+    ///
+    /// # Arguments
+    /// * `offered` - The amount that we are offering
+    /// * `requested` - The amount that we want in return
+    /// * `rng` - Randomness.
+    pub fn build_swap_proposal<T: RngCore + CryptoRng>(
+        &mut self,
+        offered: Amount,
+        requested: Amount,
+        rng: &mut T,
+    ) -> Result<SignedContingentInput> {
+        mc_common::trace_time!(self.logger, "MobileCoinClient.build_swap_proposal");
+
+        // Only one input can be used, otherwise defragmentation is required
+        let inputs = self.tx_data.get_transaction_inputs(offered, 1)?;
+        let inputs: Vec<(OwnedTxOut, TxOutMembershipProof)> = self.get_proofs(&inputs)?;
+        let rings: Vec<Vec<(TxOut, TxOutMembershipProof)>> = self.get_rings(&inputs, rng)?;
+
+        assert_eq!(inputs.len(), 1);
+        assert_eq!(rings.len(), 1);
```

Comment:
I think it's okay here because these are post conditions of the `get_transaction_data` function, so the only thing that makes sense is either to assert, or don't check it and just assume it's true. I'm okay with removing the asserts if you want that

Reply:
Ah, OK. `assert` SGTM then